src/analyze/analyze-verify-util.c

```diff
@@ -55,7 +56,7 @@ int verify_prepare_filename(const char *filename, char **ret) {
                 return -EINVAL;
 
         if (unit_name_is_valid(name, UNIT_NAME_TEMPLATE)) {
-                r = unit_name_replace_instance(name, "i", &with_instance);
+                r = unit_name_replace_instance(name, arg_instance, &with_instance);
```

Comment:
Hmm, I don't grok this? Why not just specify instance in unit id? I think the solution is to give a hint that some units might have restrictions on instance name, but no more?

Reply:
Of course, `systemd-analyze verify systemd-growfs@.service --instance=tmp` and `systemd-analyze verify systemd-growfs@tmp.service` are equivalent.

But, `--instance` may be useful for passing multiple units in a directory. E.g.
```
systemd-analyze verify /usr/lib/systemd/system/*.service --instance=hoge
```
Otherwise, users need to append instance name to each service name.

Reply:
Probably. Though I doubt that people really want to inspect so many units at once.

But apart from that, this means that the PR doesn't (intend to) resolve #33681, no?

Reply:
> Probably. Though I doubt that people really want to inspect so many units at once.

Not necessary so many, just multiple.

> But apart from that, this means that the PR doesn't (intend to) resolve #33681, no?

Maybe Closes -> Inspired by .

Reply:
Hmm, rethinking about this, maybe `test_instance` can serve as the hint well enough.

Reply:
Commit message and man page are extended.